src/Microsoft.AspNet.Mvc.Core/ModelBinders/BodyModelBinder.cs

```diff
@@ -57,7 +57,7 @@ public class BodyModelBinder : BindingSourceModelBinder
                 var unsupportedContentType = Resources.FormatUnsupportedContentType(
                     bindingContext.OperationBindingContext.HttpContext.Request.ContentType);
                 bindingContext.ModelState.AddModelError(bindingContext.ModelName, unsupportedContentType);
-                return new ModelBindingResult(null, bindingContext.ModelName, isModelSet: false);
+                return new ModelBindingResult(model: null, key: string.Empty, isModelSet: false);
```

Comment:
Lets return bindingContext.ModelName when we fail, In this case since we ourselves are adding model state error we should return that key.